frontend/apps/reader/modules/readertypeset.lua

```diff
@@ -66,12 +66,13 @@ function ReaderTypeset:onReadSettings(config)
     if config:has("copt_block_rendering_mode") then
         self.block_rendering_mode = config:readSetting("copt_block_rendering_mode")
     else
-        if config:has("last_xpointer") then
+        if config:has("last_xpointer") and not config:has("docsettings_reset_done") then
             -- We have a last_xpointer: this book was previously opened
             self.block_rendering_mode = 0
         else
             self.block_rendering_mode = G_reader_settings:readSetting("copt_block_rendering_mode")
                                      or 3 -- default to 'web' mode
+            config:delSetting("docsettings_reset_done")
```

Comment:
I would delSetting this flag in a more generic place (ie. if it's a PDF document, we won't reset it).
It might be used currently only "for readertypeset block_rendering_mode", but the idea is it should be generic and usable for other checks if later we find out we need some (ie. if needed for some other readerrolling, we don't want it delSettings the first time it is used).

Reply:
How about within
https://github.com/koreader/koreader/blob/6f45c076ecc946ce192e37a86c44bb83e128a0dd/frontend/apps/reader/modules/readerview.lua#L1003-L1005

Reply:
Sounds good.